### PR TITLE
refactor: split streaming write RPC implementation

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9556)

### CODE REVIEW

File: google/cloud/internal/streaming_write_rpc_impl.h [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#include "google/cloud/internal/streaming_write_rpc.h"
+#include "google/cloud/status.h"
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/support/sync_stream.h>
+#include <memory>
+#include <utility>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
```
Comment (devbww) [inline google/cloud/internal/streaming_write_rpc_impl.h:28]: `#include "google/cloud/version.h"`

File: google/cloud/internal/streaming_write_rpc_impl.h [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+namespace internal {
+
+/// Report the errors in a standalone function to minimize includes
```
Comment (devbww) [inline google/cloud/internal/streaming_write_rpc_impl.h:31]: I'm not sure what "in a standalone function to minimize includes" is supposed to convey.  I might instead emphasis that this a `Status` we can only surface via logging.

**Review by devbww — APPROVED** [review]

File: google/cloud/internal/streaming_write_rpc_impl.h (hunk 1 above)
Comment (coryan) [reply google/cloud/internal/streaming_write_rpc_impl.h:29]: Done

File: google/cloud/internal/streaming_write_rpc_impl.h (hunk 2 above)
Comment (coryan) [reply google/cloud/internal/streaming_write_rpc_impl.h]: Done

**Pull request merged**